packages/react/src/components/table/Table.test.tsx

```diff
@@ -0,0 +1,135 @@
+    );
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('should not have basic accessibility issues', async () => {
+    const { container } = render(
+      <Table caption={caption} cols={cols} rows={rows} indexKey={indexKey} renderIndexCol={renderIndexCol} />,
+    );
+    const results = await axe(container);
+    expect(results).toHaveNoViolations();
+  });
+
+  it('Should successfully sort the table', () => {
+    const { container } = render(
+      <Table
+        sortingEnabled
+        ariaLabelSortButtonNeutral="Not sorted"
+        ariaLabelSortButtonAscending="Sorted in ascending order"
+        ariaLabelSortButtonDescending="Sorted in descending order"
+        caption={caption}
+        cols={cols}
+        rows={rows}
+        indexKey={indexKey}
+        renderIndexCol={renderIndexCol}
+      />,
+    );
+    const ageOfFirstRow = container.querySelector('[data-testid="age-0"] > div');
+    expect(ageOfFirstRow).toHaveTextContent('39');
+    userEvent.click(container.querySelector('[data-testid="table-sorting-header-age"]'));
+    const ageOfSortedTableFirstRow = container.querySelector('[data-testid="age-0"] > div');
+    expect(ageOfSortedTableFirstRow).toHaveTextContent('8');
+  });
+
+  it('Should successfully select and deselect a single row', () => {
+    const { container } = render(
+      <Table
+        checkboxSelection
+        ariaLabelCheckboxSelection="Row selection"
+        heading={heading}
+        cols={cols}
+        rows={rows}
+        indexKey={indexKey}
+        renderIndexCol={renderIndexCol}
+      />,
+    );
+
+    expect(container.querySelector('[id="hds-table-checkbox-1000"]')).not.toBeChecked();
+
+    userEvent.click(container.querySelector('[id="hds-table-checkbox-1000"]'));
+    expect(container.querySelector('[id="hds-table-checkbox-1000"]')).toBeChecked();
+
+    userEvent.click(container.querySelector('[id="hds-table-checkbox-1000"]'));
+    expect(container.querySelector('[id="hds-table-checkbox-1000"]')).not.toBeChecked();
+  });
+
+  it('Should successfully select all and deselect all rows', () => {
+    const { container } = render(
+      <Table
+        checkboxSelection
+        ariaLabelCheckboxSelection="Row selection"
+        heading={heading}
+        cols={cols}
+        rows={rows}
+        indexKey={indexKey}
+        renderIndexCol={renderIndexCol}
+      />,
+    );
+
+    rows.forEach((row) => {
+      expect(container.querySelector(`[id="hds-table-checkbox-${row.id}"]`)).not.toBeChecked();
+    });
+
+    userEvent.click(container.querySelector('[data-testid="hds-table-select-all-button"]'));
+    rows.forEach((row) => {
+      expect(container.querySelector(`[id="hds-table-checkbox-${row.id}"]`)).toBeChecked();
+    });
+
+    userEvent.click(container.querySelector('[data-testid="hds-table-deselect-all-button"]'));
+    rows.forEach((row) => {
+      expect(container.querySelector(`[id="hds-table-checkbox-${row.id}"]`)).not.toBeChecked();
+    });
+  });
+});
```

Comment:
Could there be some test for conditional rendering, for renderIndexCol, for example?

Reply:
- [ ] Add test for conditional rendering of index column